File: leetcode/all-elements-in-two-binary-search-trees.java
// https://leetcode.com/problems/all-elements-in-two-binary-search-trees/

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {

	// Provided TreeNode class
	public class TreeNode {
		int val;
		TreeNode left;
		TreeNode right;
		TreeNode() {}
		TreeNode(int val) { this.val = val; }
		TreeNode(int val, TreeNode left, TreeNode right) {
			this.val = val;
			this.left = left;
			this.right = right;
		}
	}

    public List<Integer> getAllElementsBruteForce(TreeNode root1, TreeNode root2) {
        List<Integer> list = new ArrayList<Integer>();
        
        getAll(root1, list);
        getAll(root2, list);

        Collections.sort(list);
        
        return list;
    }
    
    private void getAll(TreeNode node, List<Integer> list){
        if(node!=null){
            list.add(node.val);
            getAll(node.left, list);
            getAll(node.right, list);
        }
    }
}

File: leetcode/all_elements_in_two_binary_search_trees.py
# https://leetcode.com/problems/all-elements-in-two-binary-search-trees


# Provided TreeNode class
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> list[int]:
        elements: list[int] = []

        self.traverse(root1, elements)
        self.traverse(root2, elements)

        elements.sort()
        return elements

    def traverse(self, node: TreeNode, elements: list[int]) -> None:
        if node:
            elements.append(node.val)
            self.traverse(node.left, elements)
            self.traverse(node.right, elements)
